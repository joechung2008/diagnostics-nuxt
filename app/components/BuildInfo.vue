<template>
  <UTable :data="data" :columns="columns">
    <template #name-data="{ row }">
      <span class="text-sm">{{ row.name }}</span>
    </template>
    <template #value-data="{ row }">
      <span class="text-sm">{{ row.value }}</span>
    </template>
  </UTable>
</template>

<script setup lang="ts">
import type { TableColumn } from "@nuxt/ui";
import type { BuildInfoProps, KeyValuePair } from "~/types";

const props = defineProps<BuildInfoProps>();

const data = computed(() => [
  {
    name: "Build Version",
    value: props.buildVersion,
  },
]);

const columns: TableColumn<KeyValuePair<string>>[] = [
  {
    accessorKey: "name",
    header: "Name",
  },
  {
    accessorKey: "value",
    header: "Value",
  },
];
</script>
