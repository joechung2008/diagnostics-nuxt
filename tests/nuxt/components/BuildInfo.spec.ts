import { mountSuspended, renderSuspended } from "@nuxt/test-utils/runtime";
import { describe, expect, it } from "vitest";
import BuildInfo from "../../../app/components/BuildInfo.vue";
import type { BuildInfoProps } from "../../../app/types";

describe("app/components/BuildInfo.vue", () => {
  it("should mount", async () => {
    const component = await mountSuspended(BuildInfo);
    expect(component).toBeDefined();
  });

  it("should render build info", async () => {
    const props: BuildInfoProps = {
      buildVersion: "1.2.3",
    };

    const { html } = await renderSuspended(BuildInfo, {
      props,
    });

    expect(html()).toMatchSnapshot();
  });
});
