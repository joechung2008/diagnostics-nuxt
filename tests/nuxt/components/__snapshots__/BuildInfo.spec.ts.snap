// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`app/components/BuildInfo.vue > should render build info 1`] = `
"<div id="test-wrapper">
  <div class="relative overflow-auto">
    <table class="min-w-full overflow-clip">
      <!--v-if-->
      <thead class="relative">
        <tr class="data-[selected=true]:bg-elevated/50">
          <th data-pinned="false" scope="col" class="px-4 py-3.5 text-sm text-highlighted text-left rtl:text-right font-semibold [&amp;:has([role=checkbox])]:pe-0">Name</th>
          <th data-pinned="false" scope="col" class="px-4 py-3.5 text-sm text-highlighted text-left rtl:text-right font-semibold [&amp;:has([role=checkbox])]:pe-0">Value</th>
        </tr>
        <tr class="absolute z-[1] left-0 w-full h-px bg-(--ui-border-accented)"></tr>
      </thead>
      <tbody class="divide-y divide-default [&amp;>tr]:data-[selectable=true]:hover:bg-elevated/50 [&amp;>tr]:data-[selectable=true]:focus-visible:outline-primary">
        <tr data-selected="false" data-selectable="false" data-expanded="false" class="data-[selected=true]:bg-elevated/50">
          <td data-pinned="false" class="p-4 text-sm text-muted whitespace-nowrap [&amp;:has([role=checkbox])]:pe-0">Build Version</td>
          <td data-pinned="false" class="p-4 text-sm text-muted whitespace-nowrap [&amp;:has([role=checkbox])]:pe-0">1.2.3</td>
        </tr>
        <!--v-if-->
      </tbody>
      <!--v-if-->
    </table>
  </div>
</div>"
`;
